src/ResourceManager/LocationBasedServices/Commands.LocationBasedServices/LocationBasedServicesAccount/LocationBasedServicesAccountBaseCmdlet.cs

```diff
@@ -0,0 +1,172 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common;
+using Microsoft.Azure.Management.LocationBasedServices;
+using Microsoft.Azure.Management.LocationBasedServices.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.LocationBasedServices
+{
+    public abstract class LocationBasedServicesAccountBaseCmdlet : AzureRMCmdlet
+    {
+        private ClientWrapper locationBasedServicesClientWrapper;
+
+        protected const string LocationBasedServicesAccountNounStr = "AzureRmLocationBasedServicesAccount";
+        protected const string LocationBasedServicesAccountKeyNounStr = LocationBasedServicesAccountNounStr + "Key";
+        protected const string LocationBasedServicesAccountSkusNounStr = LocationBasedServicesAccountNounStr + "Skus";
+
+        protected const string LocationBasedServicesAccountNameAlias = "LocationBasedServicesAccountName";
+        protected const string AccountNameAlias = "AccountName";
+
+        protected const string TagsAlias = "Tags";
+        
+        protected struct AccountSkuString 
+        {
+            internal const string S0 = "S0";
+        }
+        
+        public IClient LocationBasedServicesClient
+        {
+            get
+            {
+                if (locationBasedServicesClientWrapper == null)
+                {
+                    locationBasedServicesClientWrapper = new ClientWrapper(DefaultProfile.DefaultContext)
+                    {
+                        VerboseLogger = WriteVerboseWithTimestamp,
```

Comment:
You should update the loggers every time you get the Client.  (i.e. set these proeprties after the if block)

Reply:
ok.   i'm not sure i understand why though.   what can cause the difference?